pkg/controller/job/jobcontroller.go

```diff
@@ -419,7 +424,7 @@ func (jm *JobController) syncJob(key string) error {
 		job.Status.Failed = failed
 
 		if err := jm.updateHandler(&job); err != nil {
-			glog.Errorf("Failed to update job %v, requeuing.  Error: %v", job.Name, err)
+			utilruntime.HandleError(fmt.Errorf("Failed to update job %v, requeuing.  Error: %v", job.Name, err))
```

Comment:
Yes, but I wonder if [key](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/job/jobcontroller.go#L297) is same with [jobKey](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/job/jobcontroller.go#L327)? Since `jm.queue.AddRateLimited(jKey)` will enqueue the [former](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/job/jobcontroller.go#L297).

If the same, why we do controller.keyFunc [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/job/jobcontroller.go#L327)?